### PR TITLE
Increase CQ size to 3*MAX_REQUESTS

### PR DESCRIPTION
**Pull request by wenkaidu**
Suggested by Rukhsana Ansari <rukhsana.ansari@broadcom.com>

## Details
___Do not mention proprietary info or link to internal work items in this PR.___

**Work item:** _"Internal", or link to GitHub issue (if applicable)._
Internal

**What were the changes?**  
Increase CQ size to 3*MAX_REQUESTS

**Why were the changes made?**  
Rukhsana Ansari <rukhsana.ansari@broadcom.com> pointed out that there aren't enough CQ to handle all send and recv requests

**How was the outcome achieved?**  
Increase CQ size to 3*MAX_REQUESTS

**Additional Documentation:**  
_What else should the reviewer know?_

## Approval Checklist
___Do not approve until these items are satisfied.___
- [ ] Verify the CHANGELOG has been updated, if
  - there are any NCCL API version changes,
  - any changes impact library users, and/or
  - any changes impact any other ROCm library.

### CODE REVIEW

File: src/transport/net_ib.cc [hunk 1]
```diff
@@ -942,8 +942,8 @@ ncclResult_t ncclIbInitCommDevBase(int ibDevN, struct ncclIbNetCommDevBase* base
   base->pd = ibDev->pd;
   pthread_mutex_unlock(&ibDev->lock);
 
-  // Recv requests can generate 2 completions (one for the post FIFO, one for the Recv).
-  NCCLCHECK(wrap_ibv_create_cq(&base->cq, ibDev->context, 2*MAX_REQUESTS*ncclParamIbQpsPerConn(), NULL, NULL, 0));
+  // Recv requests can generate 3 completions (two for Send, one for the Recv).
```
Comment (rukamsari) [inline src/transport/net_ib.cc]: I think this comment needs to re-worded: 
CQ  is  sized to accommodate the max SQ + RQ WQE completions. If each SQ WQE could be signaled, then, for each QP, there can be  2*MAX_REQUESTS   completions for  SQ and  MAX_REQUESTS  completions for RQ.

Comment (wenkaidu) [reply src/transport/net_ib.cc]: updated comment

**Review by rukamsari — APPROVED** [review]

**Review by PedramAlizadeh — APPROVED** [review]

**Review by mustafabar — COMMENTED** [review]
Do we need to check if this exceeds the dev_cap.max_cqe since we are adding 50% more CQ requirements?

**Review by mustafabar — COMMENTED** [review]
I can see the max cqe is 1048575 on my end for the Thor adapter, so we probably shouldn't worry too much

**Pull request merged**